sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -5,183 +5,116 @@
+        updatedOptions: Options;
+    };
+    withContext<CallbackArgs extends unknown[], Callback extends (...args: CallbackArgs) => ReturnType<Callback>>(context: TracingContext, callback: Callback, callbackThis?: ThisParameterType<Callback>, ...callbackArgs: CallbackArgs): ReturnType<Callback>;
+    withSpan<Options extends {
+        tracingOptions?: OperationTracingOptions;
+    }, Callback extends (updatedOptions: Options, span: Omit<TracingSpan, "end">) => ReturnType<Callback>>(name: string, callback: Callback, operationOptions?: Options, spanOptions?: TracingSpanOptions, callbackThis?: ThisParameterType<Callback>): Promise<ReturnType<Callback>>;
 }
 
 // @public
-export interface OperationTracingOptions {
-    tracingContext?: Context;
+export interface TracingClientOptions {
+    namespace: string;
+    packageInformation: {
+        name: string;
+        version?: string;
+    };
 }
 
 // @public
-export function setSpan(context: Context, span: Span): Context;
-
-// @public
-export function setSpanContext(context: Context, spanContext: SpanContext): Context;
-
-// @public
-export interface Span {
-    addEvent(name: string, attributesOrStartTime?: SpanAttributes | TimeInput, startTime?: TimeInput): this;
-    end(endTime?: TimeInput): void;
-    isRecording(): boolean;
-    recordException(exception: Exception, time?: TimeInput): void;
-    setAttribute(key: string, value: SpanAttributeValue): this;
-    setAttributes(attributes: SpanAttributes): this;
-    setStatus(status: SpanStatus): this;
-    spanContext(): SpanContext;
-    updateName(name: string): this;
+export interface TracingContext {
+    deleteValue(key: symbol): TracingContext;
+    getValue(key: symbol): unknown;
+    setValue(key: symbol, value: unknown): TracingContext;
 }
 
 // @public
-export interface SpanAttributes {
-    [attributeKey: string]: SpanAttributeValue | undefined;
+export interface TracingSpan {
+    end(): void;
+    isRecording(): boolean;
+    setAttribute(name: string, value: unknown): void;
+    setStatus(status: SpanStatus): void;
+    readonly tracingSpanId: TracingSpanIdentifier;
 }
 
 // @public
-export type SpanAttributeValue = string | number | boolean | Array<null | undefined | string> | Array<null | undefined | number> | Array<null | undefined | boolean>;
-
-// @public
-export interface SpanContext {
+export interface TracingSpanIdentifier {
     spanId: string;
     traceFlags: number;
     traceId: string;
-    traceState?: TraceState;
+    traceState?: unknown;
 }
 
 // @public
-export enum SpanKind {
-    CLIENT = 2,
-    CONSUMER = 4,
-    INTERNAL = 0,
-    PRODUCER = 3,
-    SERVER = 1
-}
+export type TracingSpanKind = "client" | "server" | "producer" | "consumer" | "internal";
 
 // @public
-export interface SpanOptions {
-    attributes?: SpanAttributes;
-    kind?: SpanKind;
-    links?: Link[];
-    startTime?: TimeInput;
+export interface TracingSpanOptions {
+    spanAttributes?: {
+        [key: string]: unknown;
```

Comment:
we'd want to support string and long (int64) attributes (e.g. http status code is int)

Reply:
Yup, spanAttributes supports just about anything - `unknown` means you can pass anything you want. That seems to be how OTel expects it, although they have some smarts to ensure the type is never null or undefined. 

In our case, we will accept everything and anything, but will skip setting any attributes that are null or undefined - so string and long are both supported here (as well as an array of strings, array of long, etc)